dynomite/src/lib.rs

```diff
@@ -274,39 +308,94 @@ pub trait Attribute: Sized {
     fn from_attr(value: AttributeValue) -> Result<Self, AttributeError>;
 }
 
+impl Attribute for AttributeValue {
+    fn into_attr(self: Self) -> AttributeValue {
+        self
+    }
+    fn from_attr(value: AttributeValue) -> Result<Self, AttributeError> {
+        Ok(value)
+    }
+}
+
 /// A type capable of being produced from
 /// a set of string keys and `AttributeValues`
 pub trait FromAttributes: Sized {
+    /// Shortcut for `FromAttributes::from_mut_attrs(&mut attrs)`.
+    /// You should generally implement only that method.
+    fn from_attrs(mut attrs: Attributes) -> Result<Self, AttributeError> {
+        Self::from_mut_attrs(&mut attrs)
+    }
+
     /// Returns an instance of of a type resolved at runtime from a collection
-    /// of a `String` keys and `AttributeValues`. If
-    /// a instance can not be resolved and `AttributeError` will be returned.
-    fn from_attrs(attrs: Attributes) -> Result<Self, AttributeError>;
+    /// of a `String` keys and `AttributeValues`.
+    /// If an instance can not be resolved and `AttributeError` will be returned.
+    /// The implementations of this method should remove the relevant key-value
+    /// pairs from the map to consume them. This is needed to support
```

Comment:
It might be slightly confusing to have two methods with similar names whose arguments are both mut. Can we just make the others methods arg a borrowed mut to keep the simpler name and surface area small.

Reply:
That would be great. I'm erring on merging sooner so that could be cleanup before the next release